### DISCUSSION TITLE
Multi-agent conversations

### DISCUSSION THREAD

**Discussion by SlistInc** [Maximilian-Winter/llama-cpp-agent#6] [Q&A]
First of all: great job. I really like this project and the direction it is heading.

I was playing around with it but being didn't manage to build a setup like the following:

- I would like to have a series of agents discussing a topic
- Their order should be more or less unstructured
- They should at some point terminate the discussion and summarise points of agreement into a document

I'm fascinated by agents taking on different viewpoints. For instance:
- Have a product developer, a critic, an investor and a customer developing a product. Each defending their individual point of view and concerns.
- Have two opposing politicians and affected citizens discuss a difficult situation, trying to find a solution (with maybe a researcher finding facts to support either of them).
- Have a budgeting agent, a demanding bride, an easygoing groom, a guest and a wedding planner discussing a wedding that satisfies all sides.

I think you get the drift. There is tremendous value in having an LLM take opposing views and trying to find a common ground.

I wasn't able to find out how to get that working in your framework. Somehow there never was a "discussion" going on. Does anyone know a basic setup to get such a scenario working?

**Comment by Maximilian-Winter** [comment]
The only code I have for this was a test by me. But I will try to implement something next. Thank you!
